src/compiler/checker.ts

```diff
@@ -14963,10 +14963,12 @@ namespace ts {
                 }
             }
             else {
-                let types: Type[];
+                let yieldTypes: Type[];

```

Comment:
nit. i would leave the outside be `types`. inside `if (functionFlags & FunctionFlags.Generator)` i would just say `types = concatenate(checkAndAggregateYieldOperandTypes(func, contextualMapper), checkAndAggregateReturnExpressionTypes(func, contextualMapper));` then the keep the rest of the flow the way it was